File: momentra/momentra/Shell/Components/EditMomentSetupHost.swift
import SwiftUI

/// Resolves and hosts the existing family setup wizard for Manage Moment → Edit setup.
enum EditMomentSetupTarget: Identifiable, Equatable {
    case personal(PersonalSetupSystem)
    case business(BusinessSetupKind)
    case groupExperience
    case groupPurchase
    case groupLiving

    var id: String {
        switch self {
        case .personal(let s): return "personal:\(s.rawValue)"
        case .business(let k): return "business:\(k.rawValue)"
        case .groupExperience: return "group:experience"
        case .groupPurchase: return "group:purchase"
        case .groupLiving: return "group:living"
        }
    }

    static func resolve(
        context: AppContextKind,
        momentTypeCode: String?
    ) -> EditMomentSetupTarget? {
        switch context {
        case .personal:
            switch PersonalPulseFamily.forTypeCode(momentTypeCode) {
            case .lifeOperations: return .personal(.lifeOperations)
            case .futureBuilding: return .personal(.futureBuilding)
            case .lifestyle: return .personal(.lifestyle)
            case .relationships: return .personal(.relationships)
            }
        case .business:
            let theme = BusinessActiveTheme.forTypeCode(momentTypeCode)
            if theme.typeLabel == BusinessActiveTheme.businessRunway.typeLabel {
                return .business(.businessRunway)
            }
            if theme.typeLabel == BusinessActiveTheme.businessOperations.typeLabel {
                return .business(.businessOperations)
            }
            return .business(.teamOperations)
        case .group:
            let family = GroupExperienceFamily.forTypeCode(momentTypeCode)
            if family.isThemedPurchase { return .groupPurchase }
            if family.isThemedLiving { return .groupLiving }
            return .groupExperience
        case .circle:
            return nil
        }
    }
}

struct EditMomentSetupHost: View {
    let target: EditMomentSetupTarget
    let momentId: String
    let momentTitle: String
    let momentTypeCode: String?
    let companyId: String?
    @ObservedObject var createModel: MomentCreateModel
    var onClose: () -> Void
    var onSaved: () -> Void

    var body: some View {
        switch target {
        case .personal(let system):
            PersonalSetupWizardView(
                system: system,
                editingMomentId: momentId,
                initialTitle: momentTitle,
                onBack: onClose,
                onCreated: { _, _, _ in onSaved() }
            )
        case .business(let kind):
            if let companyId, !companyId.isEmpty {
                BusinessSetupWizardView(
                    kind: kind,
                    createModel: createModel,
                    companyId: companyId,
                    editingMomentId: momentId,
                    initialTitle: momentTitle,
                    onClose: onClose,
                    onCreated: { _ in onSaved() }
                )
            } else {
                missingCompany
            }
        case .groupExperience:
            GroupExperienceSetupView(
                createModel: createModel,
                onBack: onClose,
                onCreated: { _ in onSaved() },
                editingMomentId: momentId,
                initialTitle: momentTitle,
                initialTypeCode: momentTypeCode
            )
        case .groupPurchase:
            GroupPurchaseSetupView(
                createModel: createModel,
                onBack: onClose,
                onCreated: { _ in onSaved() },
                editingMomentId: momentId,
                initialTitle: momentTitle,
                initialTypeCode: momentTypeCode
            )
        case .groupLiving:
            GroupLivingSetupView(
                createModel: createModel,
                onBack: onClose,
                onCreated: { _ in onSaved() },
                editingMomentId: momentId,
                initialTitle: momentTitle,
                initialTypeCode: momentTypeCode
            )
        }
    }

    private var missingCompany: some View {
        VStack(spacing: 12) {
            Text("Company required")
                .font(.headline)
                .foregroundStyle(.white)
            Text("This business moment needs a company before setup can be edited.")
                .font(.subheadline)
                .foregroundStyle(.white.opacity(0.7))
                .multilineTextAlignment(.center)
            Button("Close", action: onClose)
                .buttonStyle(.borderedProminent)
        }
        .padding(24)
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .background(Color(hex: "#0C0F15"))
    }
}

File: apk/app/src/main/java/com/example/momentra/ui/shell/components/EditMomentSetupHost.kt
package com.example.momentra.ui.shell.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxHeight
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.sp
import androidx.lifecycle.viewmodel.compose.viewModel
import com.example.momentra.domain.AppContext
import com.example.momentra.ui.create.MomentCreateViewModel
import com.example.momentra.ui.shell.business.BusinessActiveTheme
import com.example.momentra.ui.shell.empty.BusinessSetupKind
import com.example.momentra.ui.shell.empty.BusinessSetupWizardContent
import com.example.momentra.ui.shell.empty.group.GroupExperienceSetupContent
import com.example.momentra.ui.shell.empty.group.GroupLivingSetupContent
import com.example.momentra.ui.shell.empty.group.GroupPurchaseSetupContent
import com.example.momentra.ui.shell.empty.personal.PersonalSetupSystem
import com.example.momentra.ui.shell.empty.personal.PersonalSetupWizardContent
import com.example.momentra.ui.shell.group.groupExperienceFamilyFor
import com.example.momentra.ui.shell.group.isThemedLiving
import com.example.momentra.ui.shell.group.isThemedPurchase
import com.example.momentra.ui.shell.personal.PersonalPulseFamily
import com.example.momentra.ui.shell.personal.personalPulseFamilyFor

@Composable
fun EditMomentSetupHost(
    context: AppContext,
    momentId: String,
    momentTitle: String,
    momentTypeCode: String?,
    companyId: String?,
    onClose: () -> Unit,
    onSaved: () -> Unit,
    createViewModel: MomentCreateViewModel = viewModel(),
) {
    val createState by createViewModel.state.collectAsState()
    Box(
        modifier = Modifier
            .fillMaxWidth()
            .fillMaxHeight(0.94f)
            .background(Color(0xFF0C0F15)),
    ) {
        when (context) {
            AppContext.PERSONAL -> {
                val system = when (personalPulseFamilyFor(momentTypeCode)) {
                    PersonalPulseFamily.LIFE_OPERATIONS -> PersonalSetupSystem.LIFE_OPERATIONS
                    PersonalPulseFamily.FUTURE_BUILDING -> PersonalSetupSystem.FUTURE_BUILDING
                    PersonalPulseFamily.LIFESTYLE -> PersonalSetupSystem.LIFESTYLE
                    PersonalPulseFamily.RELATIONSHIPS -> PersonalSetupSystem.RELATIONSHIPS
                }
                PersonalSetupWizardContent(
                    system = system,
                    onBack = onClose,
                    onCreated = { _, _, _ -> onSaved() },
                    createViewModel = createViewModel,
                    editingMomentId = momentId,
                    initialTitle = momentTitle,
                )
            }
            AppContext.BUSINESS -> {
                val theme = BusinessActiveTheme.forTypeCode(momentTypeCode)
                val kind = when {
                    theme.typeLabel == BusinessActiveTheme.BusinessRunway.typeLabel ->
                        BusinessSetupKind.BUSINESS_RUNWAY
                    theme.typeLabel == BusinessActiveTheme.BusinessOperations.typeLabel ->
                        BusinessSetupKind.BUSINESS_OPERATIONS
                    else -> BusinessSetupKind.TEAM_OPERATIONS
                }
                if (companyId.isNullOrBlank()) {
                    Text(
                        "Company required to edit business setup",
                        color = Color.White,
                        fontSize = 14.sp,
                        modifier = Modifier.align(Alignment.Center),
                    )
                } else {
                    BusinessSetupWizardContent(
                        kind = kind,
                        companyId = companyId,
                        createViewModel = createViewModel,
                        submitting = createState.submitting,
                        error = createState.error,
                        onClose = onClose,
                        onCreated = { onSaved() },
                        editingMomentId = momentId,
                        initialTitle = momentTitle,
                    )
                }
            }
            AppContext.GROUP -> {
                val family = groupExperienceFamilyFor(momentTypeCode)
                when {
                    family.isThemedPurchase() -> GroupPurchaseSetupContent(
                        onBack = onClose,
                        onCreated = { onSaved() },
                        createViewModel = createViewModel,
                        submitting = createState.submitting,
                        error = createState.error,
                        editingMomentId = momentId,
                        initialTitle = momentTitle,
                        initialTypeCode = momentTypeCode,
                    )
                    family.isThemedLiving() -> GroupLivingSetupContent(
                        onBack = onClose,
                        onCreated = { onSaved() },
                        createViewModel = createViewModel,
                        submitting = createState.submitting,
                        error = createState.error,
                        editingMomentId = momentId,
                        initialTitle = momentTitle,
                        initialTypeCode = momentTypeCode,
                    )
                    else -> GroupExperienceSetupContent(
                        onBack = onClose,
                        onCreated = { onSaved() },
                        createViewModel = createViewModel,
                        submitting = createState.submitting,
                        error = createState.error,
                        editingMomentId = momentId,
                        initialTitle = momentTitle,
                        initialTypeCode = momentTypeCode,
                    )
                }
            }
            AppContext.CIRCLE -> onClose()
        }
    }
}
